### DISCUSSION TITLE
Is Microsoft really committed to Docfx?

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/docfx#8519] [General]
I hesitate to open a discussion with the above title but based on the content of the Github discussions and very many open issues in this project, it feels like amongst the community of Docfx users there appears to be a serious concern about Microsoft's commitment to the project.

Given the excellent quality of Microsoft development tools (VS 2022, VS Code, etc) it feels the task of generating API documentation (a critical activity for many) isn't receiving the same level of care and attention.  Whilst I understand the need to split off v3 for MS Learn, this seems like a violation of the "eating your own dogfood" principle that Microsoft has lived and breathed for more than 3 decades.  The community is then left with an offering which is clearly inadequately resourced, can often be challenging to get working well and is at best opaque in its workings.  Furthermore, the detailed documentation is insufficient much of the time (see issue [#8336](https://github.com/dotnet/docfx/discussions/8336) on the subject, for related comments).

This post should not be taken as a criticism per se of Docfx; I have seen some truly world-class documentation sites built with it, for example Google's [.NET Cloud Client Libraries](https://cloud.google.com/dotnet/docs/reference/) by @jskeet and team.  Nor it is intended in any way as a criticism of the individuals working on Docfx; I am sure @yufeih and colleagues are doing their utmost to move the thing forward.

Rather, this post is more a rallying cry - that **Microsoft puts the necessary resources into Docfx to make it a first-class citizen of the .NET build toolchain**, as right now, it feels a bit like someone's side-project within the business.

PS Conscious this is an open source project, and some may say, if it bothers you that much, then please contribute.  Honestly, I would if I had the bandwidth, but as a team of 1 already working full-time on a significant open source initiative, there just aren't enough hours in the day.

**Comment by yufeih** [comment]
Thank you @paytools-master. I really appreciate your rallying cry.

Your observation is accurate. There is no resourcing from MS Learn, who previously fund this project, in continue supporting it, so we are in the process of [transition to .NET foundation](https://github.com/dotnet-foundation/projects/issues/257), hoping that it'll continue to evolve under an open source operating model.

**Reply by ghost** [reply]
Understood Yufei and I am pleased my post was helpful.  But with the deepest respect, I think Microsoft are letting the side down here.  In my view Docfx should be a properly supported part of the .NET developer ecosystem.  Feel free to raise with management - I suspect others will support this viewpoint in due course.
